ci/jenkins.build-and-test.bash

```diff
@@ -12,4 +12,4 @@ BASH_INCLUDE="$BASE_DIR/lib/bash_include"
 $THIS_DIR/clean || exit_with_error_message "clean FAILED" # DEBUG_HACK: For faster debugging when running this repeately on a dev workstation for DEBUGGING, you may want to comment this line out.
 $THIS_DIR/build || exit_with_error_message "build FAILED"
 $THIS_DIR/test  || exit_with_error_message "test FAILED"
-$THIS_DIR/check_for_newer_dependencies || { echo "WARNING: Setting jenkins job to unstable to trigger email for new dependency versions that this build has not adopted yet..." >&2; exit 99; }
+$THIS_DIR/check_for_newer_dependencies || { echo "WARNING: Setting build output code to 99 for new dependency versions that this build has not adopted yet..." >&2; exit 99; }
```

Comment:
Oh that's a great point - I hadn't realized this was a Jenkins-specific file.  I totally agree that the Jenkins references should go.  They were mainly left as a holdover in case we had issues with the Jenkins > GHA migration, but given that's done I'll start removing these.